ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSfilemanip.cpp

```diff
@@ -107,92 +114,454 @@ void ini_section_delete(std::string section)
-        default:
-            return 1;
-            break;
+int directory_create(std::string dname)
+{
+    std::string str(dname);
+    std::replace(str.begin(),str.end(),'/','\\');
+
+    if (!str.empty())
+    {
+        while (*str.rbegin() == '\\')
+        {
+            str.erase(str.size()-1);
+        }
+    }
+
+    struct stat sb;
+    if (stat((char *)str.c_str(),&sb) != 0)
+    {
+        return (_mkdir((char *)str.c_str()) == 0);
+    }
+
+    return 0;
+}
+
+int directory_destroy(std::string dname)
+{
+    std::string str(dname);
+    std::replace(str.begin(),str.end(),'/','\\');
+
+    if (!str.empty())
+    {
+        while (*str.rbegin() == '\\')
+        {
+            str.erase(str.size()-1);
+        }
+    }
+
+    struct stat sb;
+    if (stat((char *)str.c_str(),&sb) == 0 &&
+        S_ISDIR(sb.st_mode))
+    {
+        HANDLE hFind;
+        WIN32_FIND_DATA FindFileData;
+
+        TCHAR DirPath[MAX_PATH];
+        TCHAR FileName[MAX_PATH];
+
+        _tcscpy(DirPath,(char *)str.c_str());
+        _tcscat(DirPath,"\\*");
+        _tcscpy(FileName,(char *)str.c_str());
+        _tcscat(FileName,"\\");
+
+        hFind = FindFirstFile(DirPath,&FindFileData);
+        if (hFind == INVALID_HANDLE_VALUE) return 0;
+        _tcscpy(DirPath,FileName);
+
+        bool bSearch = true;
+        while (bSearch)
+        {
+            if (FindNextFile(hFind,&FindFileData))
+            {
+                if (!(_tcscmp(FindFileData.cFileName,".") &&
+                    _tcscmp(FindFileData.cFileName,".."))) continue;
+                _tcscat(FileName,FindFileData.cFileName);
+                if ((FindFileData.dwFileAttributes &
+                    FILE_ATTRIBUTE_DIRECTORY))
+                {
+                    if (!directory_destroy(FileName))
+                    {
+                        FindClose(hFind);
+                        return 0;
+                    }
+                    RemoveDirectory(FileName);
+                    _tcscpy(FileName,DirPath);
+                }
+                else
+                {
+                    if (FindFileData.dwFileAttributes &
+                        FILE_ATTRIBUTE_READONLY)
+                        _chmod(FileName, _S_IWRITE);
+
+                    if (!DeleteFile(FileName))
+                    {
+                        FindClose(hFind);
+                        return 0;
+                    }
+                    _tcscpy(FileName,DirPath);
+                }
+            }
+            else
+            {
+                if (GetLastError() == ERROR_NO_MORE_FILES)
+                    bSearch = false;
+                else
+                {
+                    FindClose(hFind);
+                    return 0;
+                }
+            }
+        }
+        FindClose(hFind);
+
+        return (RemoveDirectory((char *)str.c_str()) == true);
+    }
+
+    return 0;
+}
+
+int directory_rename(std::string oldname,std::string newname)
+{
+    std::string str1(oldname);
```

Comment:
Like in your other pull request, you're handling a lot of things here that Windows knows how to handle. It's not going to explode and delete all your files if you ask to move `my/directory/path` to `my/directory/path`, or even `my/directory/path` to `my\directory\path`. It knows how to deal with that. And also, we can't use `rename`, here. Just make a single call to `MoveFileA`/`MoveFileW`.